Allow nested structure for accordions

MartijnCuppens opened the pull request:
Fixes #25119

MartijnCuppens committed:
Allow nested structure for accordions

XhmikosR committed:
Merge branch 'v4-dev' into collapse-fix-parent

Johann-S reviewed (changes requested):
Please add a unit test

MartijnCuppens committed:
Allow nested structure for accordions

This commit allows nested structures for accordions. Also a part of
the documentation about data-children is removed because this
functionality didn't work and it's not applicable anymore.

Tests with the collapse accordion are also a bit adjusted to the new
situation.

MartijnCuppens committed:
Merge branch 'collapse-fix-parent' of github.com:MartijnCuppens/bootstrap into collapse-fix-parent

MartijnCuppens committed:
Merge branch 'v4-dev' into collapse-fix-parent

MartijnCuppens committed:
Add unit test for nested elements

MartijnCuppens commented:
- Simplified the documentation (data-children attribute was not used in js and is not needed anymore)
- Added a unit test for nested elements
- Fixed existing tests where the data-parent was set on the link and not the collapsible content

XhmikosR commented on docs/4.0/components/collapse.md:
```diff
@@ -63,7 +63,7 @@ Multiple `<button>` or `<a>` can show and hide an element if they each reference
 ## Accordion example
 
-Using the [card]({{ site.baseurl }}/docs/{{ site.docs_version }}/components/card/) component, you can extend the default collapse behavior to create an accordion.
+You can extend the default collapse behavior to create an accordion.
```
Hmm, this doesn't seem right.

Johann-S replied:
Agree with @XhmikosR

XhmikosR committed:
Update collapse.md

XhmikosR committed:
Merge branch 'v4-dev' into collapse-fix-parent
